core/src/main/scala/za/co/absa/spline/harvester/converter/ExpressionConverter.scala

```diff
@@ -79,6 +79,16 @@ class ExpressionConverter(
         name = e.prettyName,
         params = getExpressionParameters(e)
       )
+    
+    case e: sparkExprssions.ListQuery =>
+      FunctionalExpression(
+        id = idGen.nextId(),
+        dataType = convertDataTypeNonNullable(e),
```

Comment:
Since this is a workaround for a specific case in Spark, I think it would be good to add a comment referencing the #700 ticket, to make it easy to find out why standard `convertDataType` is not used here.

Reply:
The rest of the PR LGTM

Reply:
> comment referencing the https://github.com/AbsaOSS/spline-spark-agent/issues/700 ticket

and a test as well. I agree such workarounds have a tendency to vanish after some refactoring and the issue would come again in some time.

Reply:
I will try to replicate and add an example for this as I cannot share actual code. I will also try to add a test if possible.

Reply:
This is a quick integration test I wrote based on your query and it seems to work in all Spark 3.x versions including 3.4.0.

What Spark version are you testing on?

```scala
/*
 * Copyright 2019 ABSA Group Limited
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


package za.co.absa.spline

import org.apache.spark.SPARK_VERSION
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should.Matchers
import za.co.absa.commons.scalatest.ConditionalTestTags.ignoreIf
import za.co.absa.commons.version.Version.VersionStringInterpolator
import za.co.absa.spline.test.fixture.SparkFixture
import za.co.absa.spline.test.fixture.spline.SplineFixture

/**
 * https://github.com/AbsaOSS/spline-spark-agent/issues/700
 */
class Github700Test extends AsyncFlatSpec
  with Matchers
  with SparkFixture
  with SplineFixture {

  it should "not throw exception on ListQuery" taggedAs ignoreIf(ver"$SPARK_VERSION" < ver"3.0.0") in
    withIsolatedSparkSession(_.enableHiveSupport) { implicit spark =>
      spark.sql("CREATE TABLE NonNullableTest (id INT NOT NULL, name STRING, age INT)")
      spark.sql("INSERT INTO NonNullableTest VALUES (1, 'John', 25), (2, 'Jane', 30), (3, 'Bob', NULL)")

      spark.sql("CREATE TABLE NonNullableTest_1 (id INT, city STRING, country STRING)")
      spark.sql("INSERT INTO NonNullableTest_1 VALUES (null, 'New York', 'USA'), (2, 'London', 'UK'), (3, 'Paris', 'France')")

      withLineageTracking { captor =>
        for {
          (plan, _) <- captor.lineageOf {
            spark.sql(
              """
                |CREATE TABLE NonNullable_final AS
                |    SELECT *
                |      FROM NonNullableTest
                |     WHERE id IN (
                |           SELECT id
                |             FROM NonNullableTest_1
                |     )
                |""".stripMargin
            )
          }
        } yield {
          plan should not be null
        }
      }
    }

}

```

Reply:
Its appearing on all versions of Spark including 3.1,3.2 and 3.3, and its giving the error on databricks version of this. The change made in spark was backported to these versions as we can clearly see in error message and the change done in Spark recently. However, it is working well on Spark3.0 as databricks probably did not backport modified spark code to this particular version.  It's interesting to see when the opensource spark versions will have this change as last minor release version is on April 13, bug was fixed on May 16 in open source spark

Reply:
Further, I also added a reference to #700 and added a unit test for this, please review if the above reasoning sounds valid